src/test/java/org/broadinstitute/hellbender/engine/ReadInputUnitTest.java

```diff
@@ -0,0 +1,147 @@
+package org.broadinstitute.hellbender.engine;
+
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+
+public final class ReadInputUnitTest extends BaseTest {
+
+    @DataProvider(name = "InvalidReadArgumentValuesDataProvider")
+    public Object[][] getInvalidReadArgumentValues() {
+        return new Object[][] {
+                { "name:file:file2" },
+                { "name:" },
+                { ":file" },
+                { ",:file" },
+                { "name,key=value=fred:file" },
+                { "name,:file"},
+                { ",key=value:file"},
+                {"name,key:file"},
+                {"name,key=:file"},
+                {"name,=value:file"},
+                {"name,=:file"},
+                { ":" },
+                { ",:" },
+                { "::" },
+                { "" },
+                { "name,key=value1,key=value2:file" },   //duplicate key
+                { "name,key=value,key=value:file" }      //duplicate key
```

Comment:
Most of these parsing tests should be on `ParsedArgument` (soon to be `TaggedArgument`) directly, instead of being duplicated across `ReadInputUnitTest` and `FeatureInputUnitTest`. The tests for `ReadInput` and `FeatureInput` can then be much simpler.